app/models/transportation_analysis.rb

```diff
@@ -1,13 +1,29 @@
 class TransportationAnalysis < ApplicationRecord
+  before_create :compute_study_selection!, :compute_study_area!
   after_create :load_data!
```

Comment:
I forget why I put this in a `after_create` hook and not a `before_create`, as it does the same thing as both of your `compute_` methods.

Reply:
Create an issue! Click the "..."!